libs/utils/trace_analysis.py

```diff
@@ -444,20 +466,22 @@ def plotTasks(self, tasks=None, signals=None):
             signals_to_plot = list(signals_to_plot.intersection(signals))
             if len(signals_to_plot) > 0:
                 axes = plt.subplot(gs[plot_id,0]);
+                axes.set_title('Task [{0:d}:{1:s}] Signals'\
+                               .format(tid, task_name));
                 plot_id = plot_id + 1
                 is_last = (plot_id == plots_count)
-                self._plotTaskSignals(df, axes, task_name, tid,
-                                      signals_to_plot, is_last)
+                self._plotTaskSignals(axes, tid, signals_to_plot, is_last)
 
             # Plot CPUs residency
             signals_to_plot = {'residencies', 'sched_overutilized'}
             signals_to_plot = list(signals_to_plot.intersection(signals))
             if len(signals_to_plot) > 0:
                 axes = plt.subplot(gs[plot_id,0]);
+                axes.set_title('Task [{0:d}:{1:s}] Residency (green: LITTLE, red: big)'\
+                               .format(tid, task_name));
```

Comment:
Why don't we use the legend in this case too? For example, if we set ythe lowest bound of ylim to -2 we could safely add a legend to the lowest right corner.

Also, we could set the y ticks labels in the following way:

```python
axes.set_yticklabels([''] + [str(n) for n in range(self.platform['cpus_count'])] + [''])
```

This will remove the ticks labels for not present CPUs (i.e. for ticks you added to make the plot look nicer). If we add, the -2 in ylim we just need to add another empty string in the first list for `set_yticklabels`.